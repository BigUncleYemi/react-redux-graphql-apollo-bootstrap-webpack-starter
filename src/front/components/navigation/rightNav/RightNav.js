// @flow

// #region imports
import React from 'react';
import RightNavButton from './rightNavButton/RightNavButton';
// #endregion

// #region flow types
export type Props = {
  rightLinks: Array<{
    link: string,
    label: string,
    view: string,
    showWhenUserAuth: boolean,
    alwaysShows: boolean,
    btnLink: { showWhenUserAuth: boolean, alwaysShows: boolean },
  }>,
  onRightNavButtonClick: (event: SyntheticEvent<>, viewName: string) => any,
  userIsAuthenticated: boolean,
  ...any,
};
// #endregion

const RightNav = ({
  rightLinks,
  onRightNavButtonClick,
  userIsAuthenticated,
}: Props) => (
  <ul className="nav navbar-nav navbar-right">
    {userIsAuthenticated
      ? rightLinks
          .filter(({ showWhenUserAuth }) => showWhenUserAuth === true)
          .map(({ showWhenUserAuth, link, view, label }, index) => {
            return (
              <RightNavButton
                key={index}
                link={link}
                label={label}
                viewName={view}
                onClick={onRightNavButtonClick}
              />
            );
          })
      : rightLinks
          .filter(
            ({ showWhenUserAuth, alwaysShows }) =>
              showWhenUserAuth === false || alwaysShows === true,
          )
          .map(({ showWhenUserAuth, link, view, label }, index) => {
            return (
              <RightNavButton
                key={index}
                link={link}
                label={label}
                viewName={view}
                onClick={onRightNavButtonClick}
              />
            );
          })}
  </ul>
);

// #region static props
RightNav.displayName = 'RightNav';
// #endregion

export default RightNav;
