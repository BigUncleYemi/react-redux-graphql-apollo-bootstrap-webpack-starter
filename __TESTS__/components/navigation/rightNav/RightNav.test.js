import React from 'react';
import RightNav from '../../../../src/app/components/navigation/rightNav/RightNav';
import renderer from 'react-test-renderer'; // needed both for snpashot testing but also to prevent errors from enzyme
import { MemoryRouter } from 'react-router';

describe('RightNav (navigation) component', () => {
  it('renders as expected', () => {
    const props = {
      onRightNavButtonClick: () => {},
      userIsAuthenticated: false,
      rightLinks: [{ link: 'link', label: 'label', viewName: '/' }],
    };
    const component = renderer
      .create(
        <div>
          <MemoryRouter>
            <RightNav {...props} />
          </MemoryRouter>
        </div>,
      )
      .toJSON();
    expect(component).toMatchSnapshot();
  });
});
