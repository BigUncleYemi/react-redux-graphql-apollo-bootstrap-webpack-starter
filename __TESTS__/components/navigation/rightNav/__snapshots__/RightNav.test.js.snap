// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RightNav (navigation) component renders as expected 1`] = `
<div>
  <ul
    className="nav navbar-nav navbar-right"
  />
</div>
`;
